javalib/src/main/scala/java/nio/ByteArrayBits.scala

```diff
@@ -0,0 +1,181 @@
+package java.nio
+
+private[nio] object ByteArrayBits {
+  def apply(array: Array[Byte], arrayOffset: Int,
+      isBigEndian: Boolean, indexMultiplier: Int = 1): ByteArrayBits =
+    new ByteArrayBits(array, arrayOffset, isBigEndian, indexMultiplier)
+}
+
+@inline
+private[nio] final class ByteArrayBits(
```

Comment:
Yes, it is for inlining. All these tuples disappear completely after optimizations. I'll add a comment, and `@inline` annotations.